sinfonier-web/app/controllers/Modules.java

```diff
@@ -383,9 +383,13 @@ public static void recheck(String id, Integer versionCode) throws SinfonierExcep
             JsonObject res = client.uploadModule(module.getId(), moduleVersion.getId());
 
             if (res.get("code") != null && res.get("code").getAsInt() == Client.STATUS_SUCCESS) {
+              // Before to save it we should synchronized the version
+              moduleVersion = ModuleVersion.findById(moduleVersion.getId());
+
               // Recheck module version

```

Comment:
Con la solución actual, hay que volver a hacer una lectura adicional de la versión, para traer las posibles modificaciones realizadas sobre el registro por el BackEnd, para a continuación actualizar el registro completo.
Lo que propone Josemi es tan solo actualizar el campo del estado, lo que evita posibles problemas y tiene un mejor rendimiento.
With the current solution, you must do a additional reading of module version, to bring any changes made on the record by the BackEnd, then upgrade to the full record.
What Josemi proposes is only update the status field, avoiding potential problems and with a better performance.